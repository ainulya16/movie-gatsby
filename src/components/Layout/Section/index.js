import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const SectionWrapper = styled.div`
  h2 {
    /* font-family: 'AIABody'; */
    font-size: calc(18px + (26 - 14) * ((100vw - 300px) / (1600 - 300)));
    line-height: calc(1.3em + (1.5 - 1.2) * ((100vw - 300px) / (1600 - 300)));
    text-transform: uppercase;
    text-align: center;
    font-weight: bolder;
    margin-bottom: 30px;
    @media (max-width: 768px) {
      margin-bottom: 20px;
    }
  }
`;

const Section = props => {
  const { title, titleClassName, titleStyle, children, ...attr } = props;
  return (
    <SectionWrapper {...attr}>
      {title && (
        <h2 className={`margin-bottom-m ${titleClassName}`} style={titleStyle}>
          {title}
        </h2>
      )}
      {children}
    </SectionWrapper>
  );
};

Section.defaultProps = {
  title: null,
  titleClassName: null,
  titleStyle: null,
  children: null,
};

Section.propTypes = {
  title: PropTypes.string,
  titleClassName: PropTypes.string,
  titleStyle: PropTypes.objectOf(PropTypes.any),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
};

export default Section;
