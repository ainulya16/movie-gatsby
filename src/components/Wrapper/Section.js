import React from 'react';
import { Row, Col } from 'antd';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const SectionWrapper = styled.div`
  min-height: 380px;

  /* Extra small devices (phones, less than 768px) */
  @media (max-width: 768px) {
    padding: 0px;
  }
  /* Small devices (tablets, 768px and up) */
  @media (min-width: 768px) and (max-width: 992px) {
    padding: 10px;
  }
  /* Medium devices (desktops, 992px and up) */
  @media (min-width: 992px) and (max-width: 1200px) {
    padding: 14px;
  }
  /* Large devices (large desktops, 1200px and up) */
  @media (min-width: 1200px) {
    padding: 14px;
  }
`;

const Section = ({ children }) => (
  <SectionWrapper>
    <Row type="flex" align="middle" justify="center">
      <Col xs={23} sm={23} md={20} lg={20} xl={20}>
        {children}
      </Col>
    </Row>
  </SectionWrapper>
);

Section.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Section;
