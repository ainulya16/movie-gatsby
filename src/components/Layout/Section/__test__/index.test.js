import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Section from '..';

describe('Component: Layout', () => {
  it('should be exist', () => {
    const wrapper = shallow(<Section />);
    expect(wrapper.exists()).toBe(true);
  });
  it('renders Section default styles', () => {
    const tree = renderer
      .create(
        <Section title="title">
          <p>Test</p>
        </Section>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
