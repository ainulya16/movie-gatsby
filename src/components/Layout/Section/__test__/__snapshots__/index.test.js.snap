// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Layout renders Section default styles 1`] = `
.c0 h2 {
  font-size: calc(18px + (26 - 14) * ((100vw - 300px) / (1600 - 300)));
  line-height: calc(1.3em + (1.5 - 1.2) * ((100vw - 300px) / (1600 - 300)));
  text-transform: uppercase;
  text-align: center;
  font-weight: bolder;
  margin-bottom: 30px;
}

@media (max-width:768px) {
  .c0 {
    padding: 20px;
  }
}

@media (max-width:768px) {
  .c0 h2 {
    margin-bottom: 20px;
  }
}

<div
  className="c0"
>
  <h2
    className="margin-bottom-m null"
    style={null}
  >
    title
  </h2>
  <p>
    Test
  </p>
</div>
`;
